src/Game/Client.java

```diff
@@ -1212,6 +1219,59 @@ public static void redrawMenuHook(
     }
   }
 
+  public static void gameClickHook(Integer lastMenuId, int idxOrDeltaX, int idxOrDeltaY) {
+    ItemAction action = itemActionMap.get(lastMenuId);
+    Object res;
+    int mx, my;
+    if (action != null) {
+      if (action.containsWorldPoint == 1) {
+        Client.displayMenuAction(action.name, idxOrDeltaX + regionX, idxOrDeltaY + regionY);
+      } else if (action.containsWorldPoint == 2) {
+        if (Reflection.getNpc == null) return;
+        try {
+          res = Reflection.getNpc.invoke(Client.instance, idxOrDeltaX, (byte) -123);
+          mx = (Reflection.characterX.getInt(res) - 64) / Client.tileSize;
+          my = (Reflection.characterY.getInt(res) - 64) / Client.tileSize;
+          Client.displayMenuAction(action.name, mx + regionX, my + regionY);
+        } catch (Exception e) {
+        }
+      } else if (action.containsWorldPoint == 3) {
+        if (Reflection.getPlayer == null) return;
+        try {
+          res = Reflection.getPlayer.invoke(Client.instance, idxOrDeltaX, 220);
+          mx = (Reflection.characterX.getInt(res) - 64) / Client.tileSize;
+          my = (Reflection.characterY.getInt(res) - 64) / Client.tileSize;
+          Client.displayMenuAction(action.name, mx + regionX, my + regionY);
+        } catch (Exception e) {
+        }
+      }
+    }
+  }
+
+  public static void walkSourceHook(int deltaX, int deltaY) {
+    Client.displayWalkToSource(deltaX + regionX, deltaY + regionY);
+  }
+
+  public static void displayMenuAction(String action, int positionX, int positionY) {
+    Client.printAndShowActionString(action + " @ (" + positionX + "," + positionY + ")");
+  }
+
+  public static void displayMenuAction(String action, int serverId) {
+    Client.printAndShowActionString(action + " ID: " + serverId);
+  }
+
+  public static void displayWalkToSource(int positionX, int positionY) {
+    Client.printAndShowActionString("WALK_TO_SOURCE @ (" + positionX + "," + positionY + ")");
+  }
+
+  public static void printAndShowActionString(String actionString) {
+    if (Settings.SHOW_LAST_MENU_ACTION.get(Settings.currentProfile)) {
+      menu_timer = System.currentTimeMillis() + 2000L;
```

Comment:
Timer could maybe be longer, seems short.

Reply:
I guess could place it as 4-5 secs? since it also goes to console now

Reply:
3500 ms maybe lol.

Reply:
Done